src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -432,23 +432,37 @@ public BuildEnvironment(BuildEnvironmentMode mode, string currentMSBuildExePath,
             CurrentMSBuildToolsDirectory = Path.GetDirectoryName(currentMSBuildExePath);
             CurrentMSBuildConfigurationFile = string.Concat(currentMSBuildExePath, ".config");
 
+            // We can't detect an environment, don't try to set paths.
+            if (mode == BuildEnvironmentMode.None)
+                return;
+
             VisualStudioInstallRootDirectory = visualStudioPath;
 
-            if (mode == BuildEnvironmentMode.VisualStudio)
-            {
-                var isAmd64 = FileUtilities.EnsureNoTrailingSlash(CurrentMSBuildToolsDirectory)
-                    .EndsWith("amd64", StringComparison.OrdinalIgnoreCase);
+            // Check to see if our current folder is 'amd64'
+            bool runningInAmd64 = FileUtilities.EnsureNoTrailingSlash(CurrentMSBuildToolsDirectory)
+                .EndsWith("amd64", StringComparison.OrdinalIgnoreCase);
```

Comment:
Should this check for `{directoryseparator}amd64`? It'd be confused if someone had a `toolsamd64\msbuild.exe` (though that'd get caught later).